### DISCUSSION TITLE
Animating Widget Offset

### DISCUSSION THREAD

**Discussion by lllama** [Textualize/textual#1569] [Q&A]
The docs say that you can animate a Widget's `offset` (https://textual.textualize.io/guide/animation/#animating-styles) in order to move it around the screen.

At the moment, I'm only able to do this by creating a custom widget that wraps my desired widget, and then adding two reactive attributes (`offset_x` and `offset_y`). I then use the `watch` methods to change the actual offset using

```python
def watch_offset_x(self, new_value):
    self.styles.offset = ScalarOffset(Scalar.from_number(new_value), self.styles.offset.y)
```

Is there some other way that I'm missing?

**Comment by willmcgugan** [comment]
I would expect modifying offset property to be enough. Sounds like a bug.

**Reply by lllama** [reply]
Ah! So the error message says `animate` needs an `int`, `float` or something with a `blend` method. Looks like there's an `Offset` type with one of those that I haven't tried...

**Reply by willmcgugan** [reply]
What were you setting `offset` to? Even if using an Offset works, I would like it to accept a tuple of numbers as well.

**Reply by lllama** [reply]
So I might have just tried with a float and then realised that I needed something like a tuple and assumed it was a gap in the docs so went down the `reactive` path. Just tried with a `tuple` and an `Offset` though, and neither work: `tuple` complains it's not an `int`, `float`, or `blend`able thing, and the `Offset` is okay as a final value but then it errors with a exception that the `start_value` needs to be a float. i.e. it's happy getting to an `Offset` but it doesn't like `ScalarOffset` as a `start_value`.

So I think that this is a bug/missing case.
